metric/distribution/regular/regular_distribution.go

```diff
@@ -69,6 +70,9 @@ func (regularDist *RegularDistribution) Size() int {
 
 // weight is 1/samplingRate
 func (regularDist *RegularDistribution) AddEntryWithUnit(value float64, weight float64, unit string) error {
+	if math.IsNaN(value) || math.IsInf(value, 0) {
```

Comment:
Could refactor this a bit and remove a level of indentation.
func ... {
    if weight <= 0 || math.IsNaN(value) || math.IsInf(value, 0) || value < 0 {
        return fmt.Errof("E! unsupported weight or value, w %v, v %v", weight, value)
    }
    .
    .
    .
}
```